oneview_redfish_toolkit/blueprints/subscription.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2017) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+
+from flask import abort
+from flask import Blueprint
+from flask import request
+from flask import Response
+from flask_api import status
+
+from oneview_redfish_toolkit.api.errors import OneViewRedfishError
+from oneview_redfish_toolkit.api.subscription import Subscription
+from oneview_redfish_toolkit import util
+
+
+subscription = Blueprint("subscription", __name__)
+
+
+@subscription.route(
+    "/redfish/v1/EventService/EventSubscriptions/", methods=["POST"])
+def add_subscription():
+    """Add the Redfish Subscription.
+
+        Add a new subscription when this POST operation is requested.
+        The body of the request must have an array of EventTypes.
+
+        EventTypes:
+            - StatusChange
+            - ResourceUpdated
+            - ResourceAdded
+            - ResourceRemoved
+            - Alert
+
+        Returns:
+            JSON: JSON with Subscription information.
+
+        Exception:
+            OneViewRedfishError: When occur a key mapping error.
+            return abort(400)
+
+            Exception: Unexpected error.
+            return abort(500)
+    """
+    try:
+        # get information from post
+        # add subscription in subscriptions_by_type
+        # add subscription in all_subscriptions
+        # the subscription id is len(all_subscriptions) + 1
+
+        try:
+            body = request.get_json()
+            event_types = body["EventTypes"]
+        except Exception:
+            raise OneViewRedfishError(
+                {"errorCode": "INVALID_INFORMATION",
+                 "message": "Invalid JSON key. The JSON request body"
+                            " must have the key EventTypes."})
+
+        subscription_id = str(len(util.all_subscriptions) + 1)
+
+        # Build Subscription object and validates it
+        sc = Subscription(subscription_id, event_types)
+
+        for event_type in event_types:
+            if event_type not in util.subscriptions_by_type:
```

Comment:
If you use UUID, it will never be... if you use a sequential id, then you will have problems when users start deleting subscriptions, wdyt?

Reply:
The required value is an `Id`, that according the spec it is different than `UUID`. Should we generate a random `UUID`?

Reply:
We will use UUID.